sys/net/gnrc/link_layer/gomach/gomach.c

```diff
@@ -1976,7 +1976,12 @@ static int _send(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt)
 
 static void _gomach_msg_handler(gnrc_netif_t *netif, msg_t *msg)
 {
+    netdev_t *dev = netif->dev;
     switch (msg->type) {
+        case NETDEV_MSG_TYPE_EVENT:
+            dev->driver->isr(dev);
+            break;
+
```

Comment:
From https://github.com/RIOT-OS/RIOT/pull/13608#discussion_r391100830

> done!

You sure? ;-)

Reply:
ah, it's actually wrong here.
That `break` should be a `return`.

> For gomach and lwmac I would keep it as it is, since there's a check in the end that might delay ISR processing.

I'm afraid of that `while` loop in the end. I'm not sure if it's meant to be executed when processing the ISR. To keep the original behavior, that's why I decided to write it that way

Reply:
Be less afraid, just do

```C
static void _gomach_msg_handler(gnrc_netif_t *netif, msg_t *msg)
{
    gnrc_netif_msg_handler_netdev(netif, msg);
    switch (msg->type) {
        case GNRC_GOMACH_EVENT_RTT_TYPE: {
            ...
```
That's what I asked from the beginning ;-)

Reply:
On the other hand: you can also just put the `gnrc_netif_msg_handler_netdev()` into the `default:` case, right? I don't see how this would break anything.

Reply:
the problem is, with the old behavior the msg_handler was not visited if `msg->type` was NETDEV_EVENT_MSG_TYPE. So the `while` loop in the end was not called.

Reply:
I see, okay, then it is probably best to not inherit from the default handler.

Reply:
I know it doesn't look nice :(. But I also have no idea about gomach internals